src/commissaire_service/investigator/__init__.py

```diff
@@ -0,0 +1,204 @@
+            self.logger.warn(
+                'Unable to continue for {0} due to '
+                '{1}: {2}. Returning...'.format(address, type(error), error))
+            raise error
+
+        transport = ansibleapi.Transport(host.remote_user)
+
+        key = TemporarySSHKey(host, self.logger)
+        try:
+            key.create()
+        except Exception as error:
+            self.logger.warn(
+                'Unable to continue for {0} due to '
+                '{1}: {2}. Returning...'.format(address, type(error), error))
+            raise error
+
+        try:
+            facts = transport.get_info(address, key.path)
+            # recreate the host instance with new data
+            data = json.loads(host.to_json(secure=True))
+            data.update(facts)
+            host = Host(**data)
+            host.last_check = datetime.datetime.utcnow().isoformat()
+            host.status = 'bootstrapping'
+            self.logger.info('Facts for {0} retrieved'.format(address))
+            self.logger.debug('Data: {0}'.format(host.to_json()))
+        except Exception as error:
+            self.logger.warn('Getting info failed for {0}: {1}'.format(
+                address, str(error)))
+            host.status = 'failed'
+            key.remove()
+            raise error
+
+        self.logger.info(
+            'Finished and stored investigation data for {0}'.format(address))
+        self.logger.debug(
+            'Finished investigation update for {0}: {1}'.format(
+                address, host.to_json()))
+
+        self.logger.info('{0} is now in bootstrapping'.format(address))
+        oscmd = get_oscmd(host.os)
+        try:
+            etcd_config = self._get_etcd_config()
+            cluster, network = self._get_cluster_and_network_models(
+                cluster_data)
+            transport.bootstrap(
+                address, key.path, oscmd, etcd_config, cluster, network)
+            host.status = 'inactive'
+        except Exception as error:
+            self.logger.warn(
+                'Unable to start bootstraping for {0}: {1}'.format(
+                    address, str(error)))
+            host.status = 'disassociated'
+            key.remove()
+            raise error
+
+        # Verify association with relevant container managers
+        params = {
+            'cluster_type': cluster.type,
+            'address': address
+        }
+        response = self.request('storage.node_registered', params=params)
+        if response['result']:
+            host.status = 'active'
+
+        self.logger.info(
+            'Finished bootstrapping for {0}'.format(address))
+        self.logger.debug('Finished bootstrapping for {0}: {1}'.format(
+            address, host.to_json()))
+
+        # XXX TEMPORARILY DISABLED
+        # WATCHER_QUEUE.put_nowait((host, datetime.datetime.utcnow()))
+
+        key.remove()
+
+        return host.to_json()
+
+
+if __name__ == '__main__':
```

Comment:
Just noticed this while running the command. The logic here should be put into a main function so it can be referenced by ``setup.py`` entrypoints. Otherwise the CLI won't work.